mlflow/models/evaluation/base.py

```diff
@@ -957,6 +951,26 @@ def _validate(validation_thresholds, candidate_metrics, baseline_metrics=None):
     raise ModelValidationFailedException(message=os.linesep.join(failure_messages))
 
 
+def _convert_data_to_mlflow_dataset(data, targets=None, dataset_path=None):
+    """Convert input data to mlflow dataset."""
+    if "pyspark" in sys.modules:
+        from pyspark.sql import DataFrame as SparkDataFrame
+
+    if isinstance(data, Dataset):
+        return data
+    elif isinstance(data, pd.DataFrame):
+        return mlflow.data.from_pandas(df=data, targets=targets, source=dataset_path)
+    elif "pyspark" in sys.modules and isinstance(data, SparkDataFrame):
+        return mlflow.data.from_spark(df=data, targets=targets, path=dataset_path)
+    elif isinstance(data, np.ndarray):
+        return mlflow.data.from_numpy(data, targets=targets, source=dataset_path)
+    else:
+        raise TypeError(
+            "`data` must be a `mlflow.data.dataset.Dataset`, pandas DataFrame, numpy array or "
+            f"Spark DataFrame, but received `data` of type {type(data)}."
+        )
```

Comment:
i dont think we should throw here. logging the dataset as part of mlflow evaluation is sort of a "bonus" feature. i think we should gracefully handle any errors with logging a dataset (by emitting a warning to the user), but the evaluation itself should still go through. cc: @dbczumar to confirm.

Reply:
This is sanity checking if the input data type is allowed, and prompt users to change the data type if not. Putting this logic inside `_convert_data_to_mlflow_dataset` is a bit confusing, maybe we should do the sanity check in a separate method. I am doing this because we are converting every data type into Dataset, so just being lazy to write another function.

Reply:
@chenmoneygithub can we remove this for now?

Reply:
@dbczumar Yea I am down to remove the `TypeError` from this method!

I will open a separate PR to sanity check the input. Basically we need to have an explicit list of supporting input types, and throw an error on unexpected types (even if it works magically).

Reply:
@chenmoneygithub I think we should let `mlflow.pyfunc.predict()` handle the set of supported input types. I don't think we should be re-validating in `mlflow.evaluate()`. As such, I think it's best not to file that follow-up for now

Reply:
@dbczumar share my thoughts on this - Let's assume users pass in an input type of `torch.Tensor` to `mlflow.evaluate()`, then I am comparing between these 2 error messages:
1. `predict()` must have input type of ..., but received `torch.Tensor`.
2. `evaluate()` must have input type of ..., but received `torch.Tensor`.

When prompted with 1, users need to read through the stack trace to understand how the input to `predict()` is related to the input to `evaluate()`, without knowing much context, the action item isn't straighforward. When prompted with 2, users have a clear action item to fix their code.

In Tensorflow we actually did a few rounds of error message improvement, and throwing a clear error on disallowed type on public method was one part of the cleanup, as users were complaining about error messages are hidden deep in the stack.

Reply:
Sounds like it's `mlflow.evaluate()`'s job to catch the input type validation exception thrown by `predict()` and provide a user-friendly explanation of how to adjust their `evaluate()` input for compatibility with `predict()`, rather than having `mlflow.evaluate()` define its own validation logic.

Reply:
Yea, it should work too! It's not blocking, we can skip it and make a fix if we receive user reported issue.